arbnode/inbox_reader.go

```diff
@@ -39,6 +40,12 @@ func (c *InboxReaderConfig) Validate() error {
 	if c.MaxBlocksToRead == 0 || c.MaxBlocksToRead < c.DefaultBlocksToRead {
 		return errors.New("inbox reader max-blocks-to-read cannot be zero or less than default-blocks-to-read")
 	}
+	if c.ReadMode != "latest" {
+		c.ReadMode = strings.ToLower(c.ReadMode)
+		if c.ReadMode != "safe" && c.ReadMode != "finalized" {
+			return fmt.Errorf("inbox reader read-mode is invalid, want: safe or finalized, got: %s", c.ReadMode)
+		}
+	}
```

Comment:
Shouldn't it be more like:
```
		if if c.ReadMode != "latest" && c.ReadMode != "safe" && c.ReadMode != "finalized" {
			return fmt.Errorf("inbox reader read-mode is invalid, want: latest, safe, or finalized, got: %s", c.ReadMode)
		}
```

Ambivalent about the ToLower. Are we doing that elsewhere in config validation? More important to be consistent.

Reply:
"latest" is the default value so I didn't consider it in the check, but its better to include it just to be safe.

Staker uses ToLower to parse strategy and it seemed like a good idea to do here well.